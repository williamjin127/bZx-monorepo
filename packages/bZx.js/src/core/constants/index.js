export const SOLIDITY_TYPE_MAX_CHARS = 64;
