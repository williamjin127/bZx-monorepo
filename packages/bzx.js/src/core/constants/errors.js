export const ContractDoesNotExist = "Contract does not exist at address.";
export const NoNetworkId = "Missing networkId. Provide a networkId param.";
export const InvalidSignature = "Signature is invalid.";
