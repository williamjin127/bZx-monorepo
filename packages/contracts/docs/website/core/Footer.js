/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

const React = require("react");

const siteConfig = require(process.cwd() + "/siteConfig.js");

class Footer extends React.Component {
  docUrl(doc, language) {
    const baseUrl = this.props.config.baseUrl;
    return baseUrl + "docs/" + (language ? language + "/" : "") + doc;
  }

  pageUrl(doc, language) {
    const baseUrl = this.props.config.baseUrl;
    return baseUrl + (language ? language + "/" : "") + doc;
  }

  render() {
    const currentYear = new Date().getFullYear();
    return (
      <footer className="nav-footer" id="footer">
        <section className="sitemap">
          <a href={this.props.config.baseUrl} className="nav-home">
            {this.props.config.footerIcon && (
              <img
                src={this.props.config.baseUrl + this.props.config.footerIcon}
                alt={this.props.config.title}
                width="66"
                height="58"
              />
            )}
          </a>
          <div>
            <h5>Docs</h5>
            <a href={this.docUrl("migrations_Migrations.html")}>
              Getting Started
            </a>
          </div>
          <div>
            <h5>Community</h5>
            <a
              href={siteConfig.stackoverflow}
              target="_blank"
              rel="noreferrer noopener"
            >
              Stack Overflow
            </a>
            <a
              href={siteConfig.twitter}
              target="_blank"
              rel="noreferrer noopener"
            >
              Twitter
            </a>
          </div>
          <div>
            <h5>More</h5>
            <a href={siteConfig.medium} target="_blank">
              Medium
            </a>
            <a href={siteConfig.githubCompanyRepository} target="_blank">
              GitHub
            </a>
            <a
              className="github-button"
              href={siteConfig.githubRepository}
              data-icon="octicon-star"
              data-count-href="/bZxNetwork/protocol_contracts/stargazers"
              data-show-count={true}
              data-count-aria-label="# stargazers on GitHub"
              aria-label="Star this project on GitHub"
            >
              Star
            </a>
          </div>
        </section>
        <section className="copyright">{this.props.config.copyright}</section>
      </footer>
    );
  }
}

module.exports = Footer;
