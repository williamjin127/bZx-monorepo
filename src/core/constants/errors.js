export const ContractDoesNotExist = "Contract does not exist at address.";
