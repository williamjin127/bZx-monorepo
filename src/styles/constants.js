export const COLORS = {
  blue: `rgb(25, 197, 194)`,
  white: `#FFFFFF`,
  gray: `grey`
};

export const primaryPalette = {
  50: `#e5efff`,
  100: `#bfd7ff`,
  200: `#94bcff`,
  300: `#69a1ff`,
  400: `#488dff`,
  500: `#2879ff`,
  600: `#2471ff`,
  700: `#407bf7`,
  800: `#185cff`,
  900: `#0f49ff`,
  A100: `#94bcff`,
  A200: `#69a1ff`,
  A400: `#488dff`,
  A700: `#2879ff`,
  contrastDefaultColor: `light`
};

export const accentColor = `#FFD112`;

export const SHADOWS = {
  nav: `0px 2px 4px -1px rgba(0, 0, 0, 0.2),
    0px 4px 5px 0px rgba(0, 0, 0, 0.14),
    0px 1px 10px 0px rgba(0, 0, 0, 0.12)`,
  card: `0px 1px 5px 0px rgba(0, 0, 0, 0.2),
    0px 2px 2px 0px rgba(0, 0, 0, 0.14),
    0px 3px 1px -2px rgba(0, 0, 0, 0.12)`,
  light: `0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)`,
  dark: `0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23)`,
  debugRed: `0 0px 50px rgba(255, 0, 0, 0.5), 0 0px 50px rgba(255, 0, 0, 0.8)`
};

export const DIMENSIONS = {
  maxWidth: `1024px`,
  mobileNavHeight: `64px`
};
