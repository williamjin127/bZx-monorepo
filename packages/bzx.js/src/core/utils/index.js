import { BigNumber } from "@0xproject/utils";
import { assert } from "@0xproject/assert";
import BN from "bn.js";
import Web3Utils from "web3-utils";
import * as constants from "../constants";
import { SchemaValidator } from "../../schemas/bZx_json_schemas";
import { getContracts } from "../../contracts";
import * as Addresses from "../../addresses";

export const noop = () => {};

export const bigNumberToBN = value => new BN(value.toString(), 10);

export const generatePseudoRandomSalt = () => {
  // BigNumber.random returns a pseudo-random number between 0 & 1
  // with a passed in number of decimal places.
  // Source: https://mikemcl.github.io/bignumber.js/#random
  const randomNumber = BigNumber.random(
    constants.MAX_DIGITS_IN_UNSIGNED_256_INT
  );
  const factor = new BigNumber(10).pow(
    constants.MAX_DIGITS_IN_UNSIGNED_256_INT - 1
  );
  const salt = randomNumber.times(factor).round();
  return salt;
};

const getOrderValues = (order, shouldFormatAsStrings) => {
  // Must be strings in production for Web3Utils.soliditySha3 for some reason
  if (shouldFormatAsStrings) {
    return [
      order.loanTokenAmount.toString(),
      order.interestAmount.toString(),
      order.initialMarginAmount.toString(),
      order.maintenanceMarginAmount.toString(),
      order.lenderRelayFee.toString(),
      order.traderRelayFee.toString(),
      order.maxDurationUnixTimestampSec.toString(),
      order.expirationUnixTimestampSec.toString(),
      order.makerRole.toString(),
      order.withdrawOnOpen.toString(),
      order.salt.toString()
    ];
  }
  return [
    bigNumberToBN(order.loanTokenAmount),
    bigNumberToBN(order.interestAmount),
    bigNumberToBN(order.initialMarginAmount),
    bigNumberToBN(order.maintenanceMarginAmount),
    bigNumberToBN(order.lenderRelayFee),
    bigNumberToBN(order.traderRelayFee),
    bigNumberToBN(order.maxDurationUnixTimestampSec),
    bigNumberToBN(order.expirationUnixTimestampSec),
    bigNumberToBN(order.makerRole),
    bigNumberToBN(order.withdrawOnOpen),
    bigNumberToBN(order.salt)
  ];
};

const getLoanOrderHashArgs = (order, shouldFormatAsStrings) => {
  const orderAddresses = [
    order.makerAddress,
    order.loanTokenAddress,
    order.interestTokenAddress,
    order.collateralTokenAddress,
    order.feeRecipientAddress,
    order.oracleAddress,
    order.takerAddress,
    order.tradeTokenToFillAddress
  ];
  const orderValues = getOrderValues(order, shouldFormatAsStrings);
  const oracleData = order.oracleData || "0x";

  return { orderAddresses, orderValues, oracleData };
};

export const doesContractExistAtAddress = async (web3, address) => {
  const code = await web3.eth.getCode(address);
  // Regex matches 0x0, 0x00, 0x in order to accommodate poorly implemented clients
  const codeIsEmpty = /^0x0{0,40}$/i.test(code);
  return !codeIsEmpty;
};

export const getContractInstance = (web3, abi, address) => {
  assert.isETHAddressHex("address", address);
  const contract = new web3.eth.Contract(abi, address);
  return contract;
};

export const getLoanOrderHashHex = order => {
  const { orderAddresses, orderValues, oracleData } = getLoanOrderHashArgs(order, true);
  const orderHashHex = Web3Utils.soliditySha3(
    { t: "address", v: order.bZxAddress },
    { t: "address[8]", v: orderAddresses },
    { t: "uint256[11]", v: orderValues },
    { t: "bytes", v: oracleData }
  );
  return orderHashHex;
};

export const getLoanOrderHashAsync = async ({ web3, networkId }, order) => {
  const { orderAddresses, orderValues, oracleData } = getLoanOrderHashArgs(order, true);
  const bZxContract = await getContractInstance(
    web3,
    getContracts(networkId).BZx.abi,
    Addresses.getAddresses(networkId).BZx
  );
  return bZxContract.methods
    .getLoanOrderHash(
      orderAddresses,
      orderValues,
      oracleData
    )
    .call();
};

export const doesConformToSchema = (variableName, value, schema) => {
  const schemaValidator = new SchemaValidator();
  const validationResult = schemaValidator.validate(value, schema);
  const hasValidationErrors = validationResult.errors.length > 0;
  const msg = `Expected ${variableName} to conform to schema ${
    schema.id
  }\nEncountered: ${JSON.stringify(
    value,
    null,
    "\t"
  )}\nValidation errors: ${validationResult.errors.join(", ")}`;
  assert.assert(!hasValidationErrors, msg);
};

export const toChecksumAddress = addr => Web3Utils.toChecksumAddress(addr);

export const requestFaucetToken = (
  { web3, networkId },
  { tokenAddress, receiverAddress, getObject, txOpts }
) => {
  const faucetContract = getContractInstance(
    web3,
    getContracts(networkId).TestNetFaucet.abi,
    Addresses.getAddresses(networkId).TestNetFaucet
  );

  const txObj = faucetContract.methods.faucet(
    toChecksumAddress(tokenAddress),
    toChecksumAddress(receiverAddress)
  );
  console.log(`requestFaucetToken: ${txObj.encodeABI()}`);

  if (getObject) {
    return txObj;
  }
  return txObj.send(txOpts);
};

export const getWeb3Contract = ({ web3, networkId }, contractName, contractAddress) => getContractInstance(
    web3,
    getContracts(networkId)[contractName].abi,
    contractAddress ? contractAddress : Addresses.getAddresses(networkId)[contractName]
  );
