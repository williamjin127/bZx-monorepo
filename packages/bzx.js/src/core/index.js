import { assert } from "@0xproject/assert";
import { BigNumber } from "@0xproject/utils";
import * as constants from "./constants";
import { schemas } from "../schemas/bZx_json_schemas";
import * as utils from "./utils";
import * as tokenRegistry from "../tokenRegistry";
import EIP20 from "../contracts/EIP20.json";
import * as allowance from "../allowance";
import * as oracles from "../oracles";
import * as fill from "../fill";
import * as Addresses from "../addresses";
import * as orderHistory from "../orderHistory";
import * as transfer from "../transfer";
import * as signature from "../signature";
import * as Errors from "./constants/errors";
import * as trade from "../trade";
import * as loanHealth from "../loanHealth";
import * as bounty from "../bounty";
import * as weth from "../weth";

export class BZxJS {
  static generatePseudoRandomSalt = utils.generatePseudoRandomSalt;
  static noop = utils.noop;
  static toChecksumAddress = utils.toChecksumAddress;

  /* On Metamask, provider.host is undefined
  Force users to provide host url */
  constructor(web3, { networkId, addresses = Addresses.getAddresses(networkId) } = {}) {
    if (!networkId) throw new Error(Errors.NoNetworkId);

    this.web3 = web3;
    this.addresses = addresses;
    this.networkId = networkId;
    switch (networkId) {
      case 1:
        this.networkName = "mainnet";
        this.etherscanURL = "https://etherscan.io/";
        break;
      case 3:
        this.networkName = "ropsten";
        this.etherscanURL = "https://ropsten.etherscan.io/";
        break;
      case 4:
        this.networkName = "rinkeby";
        this.etherscanURL = "https://rinkeby.etherscan.io/";
        break;
      case 42:
        this.networkName = "kovan";
        this.etherscanURL = "https://kovan.etherscan.io/";
        break;
      default:
        this.networkName = "local";
        this.etherscanURL = "";
        break;
    }
  }

  static getLoanOrderHashHex(order) {
    utils.doesConformToSchema("loanOrder", order, schemas.loanOrderSchema);
    const orderHashHex = utils.getLoanOrderHashHex(order);
    return orderHashHex;
  }
  getLoanOrderHashAsync = async props =>
    utils.getLoanOrderHashAsync(this, props);

  static isValidSignature = props => signature.isValidSignature(props);

  isValidSignatureAsync = async props =>
    signature.isValidSignatureAsync(this, props);

  signOrderHashAsync = async (...props) =>
    signature.signOrderHashAsync(this, ...props);

  setAllowance = (...props) => allowance.setAllowance(this, ...props);

  setAllowanceUnlimited = props =>
    this.setAllowance({
      ...props,
      amountInBaseUnits: constants.UNLIMITED_ALLOWANCE_IN_BASE_UNITS
    });

  resetAllowance = props =>
    this.setAllowance({
      ...props,
      amountInBaseUnits: new BigNumber(0)
    });

  getAllowance = async (...props) => allowance.getAllowance(this, ...props);

  getBalance = async ({ tokenAddress, ownerAddress }) => {
    assert.isETHAddressHex("ownerAddress", ownerAddress);
    assert.isETHAddressHex("tokenAddress", tokenAddress);

    const tokenContract = await utils.getContractInstance(
      this.web3,
      EIP20.abi,
      tokenAddress
    );
    const balance = await tokenContract.methods.balanceOf(ownerAddress).call();
    return new BigNumber(balance);
  };

  getTokenList = async () => tokenRegistry.getTokenList(this);

  getOracleList = async () => oracles.getOracleList(this);
  isTradeSupported = async (...props) =>
    oracles.isTradeSupported(this, ...props);
  getConversionData = async (...props) =>
    oracles.getConversionData(this, ...props);

  takeLoanOrderAsLender = (...props) =>
    fill.takeLoanOrderAsLender(this, ...props);

  takeLoanOrderAsTrader = (...props) =>
    fill.takeLoanOrderAsTrader(this, ...props);

  pushLoanOrderOnChain = (...props) =>
    fill.pushLoanOrderOnChain(this, ...props);

  takeLoanOrderOnChainAsTrader = (...props) =>
    fill.takeLoanOrderOnChainAsTrader(this, ...props);

  takeLoanOrderOnChainAsLender = (...props) =>
    fill.takeLoanOrderOnChainAsLender(this, ...props);

  cancelLoanOrder = (...props) => fill.cancelLoanOrder(this, ...props);

  cancelLoanOrderWithHash = (...props) =>
    fill.cancelLoanOrderWithHash(this, ...props);

  getInitialCollateralRequired = async (...props) =>
    fill.getInitialCollateralRequired(this, ...props);

  orderFilledAmount = async (...props) =>
    fill.orderFilledAmount(this, ...props);
  orderCancelledAmount = async (...props) =>
    fill.orderCancelledAmount(this, ...props);

  getSingleOrder = async (...props) =>
    orderHistory.getSingleOrder(this, ...props);
  getOrdersFillable = async (...props) =>
    orderHistory.getOrdersFillable(this, ...props);
  getOrdersForUser = async (...props) =>
    orderHistory.getOrdersForUser(this, ...props);
  getSingleLoan = async (...props) =>
    orderHistory.getSingleLoan(this, ...props);
  getLoansForLender = async (...props) =>
    orderHistory.getLoansForLender(this, ...props);
  getLoansForTrader = async (...props) =>
    orderHistory.getLoansForTrader(this, ...props);

  transferToken = (...props) => transfer.transferToken(this, ...props);

  tradePositionWith0x = (...props) => trade.tradePositionWith0x(this, ...props);
  tradePositionWith0xV2 = (...props) => trade.tradePositionWith0xV2(this, ...props);
  tradePositionWithOracle = (...props) =>
    trade.tradePositionWithOracle(this, ...props);

  depositCollateral = (...props) => loanHealth.depositCollateral(this, ...props);
  withdrawCollateral = (...props) => loanHealth.withdrawCollateral(this, ...props);
  changeCollateral = (...props) => loanHealth.changeCollateral(this, ...props);
  withdrawPosition = (...props) => loanHealth.withdrawPosition(this, ...props);
  depositPosition = (...props) => loanHealth.depositPosition(this, ...props);
  getPositionOffset = (...props) => loanHealth.getPositionOffset(this, ...props);

  closeLoanPartially = (...props) => loanHealth.closeLoanPartially(this, ...props);
  closeLoan = (...props) => loanHealth.closeLoan(this, ...props);

  payInterestForOracle = (...props) => loanHealth.payInterestForOracle(this, ...props);
  getLenderInterestForOracle = (...props) => loanHealth.getLenderInterestForOracle(this, ...props);
  getTraderInterestForLoan = (...props) => loanHealth.getTraderInterestForLoan(this, ...props);

  requestFaucetToken = (...props) => utils.requestFaucetToken(this, ...props);

  getActiveLoans = (...props) => bounty.getActiveLoans(this, ...props);
  getMarginLevels = (...props) => bounty.getMarginLevels(this, ...props);
  liquidateLoan = (...props) => bounty.liquidateLoan(this, ...props);

  wrapEth = (...props) => weth.wrapEth(this, ...props);
  unwrapEth = (...props) => weth.unwrapEth(this, ...props);

  getWeb3Contract = (...props) => utils.getWeb3Contract(this, ...props);
}

export default BZxJS;
