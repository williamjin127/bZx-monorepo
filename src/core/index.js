import { assert } from "@0xproject/assert";
import { constants } from "0x.js/lib/src/utils/constants";
import { BigNumber } from "@0xproject/utils";
import { schemas } from "../schemas/bZx_json_schemas";
import * as utils from "./utils";
import * as tokenRegistry from "../tokenRegistry";
import EIP20 from "../contracts/EIP20.json";
import * as allowance from "../allowance";
import * as oracles from "../oracles";
import * as fill from "../fill";
import * as Addresses from "../addresses";
import * as orderHistory from "../orderHistory";
import * as transfer from "../transfer";
import * as signature from "../signature";
import * as Errors from "./constants/errors";
import * as trade from "../trade";
import * as loanHealth from "../loanHealth";
import * as bounty from "../bounty";
import * as weth from "../weth";

const Web3 = require("web3"); // eslint-disable global-require

export default class BZxJS {
  static generatePseudoRandomSalt = utils.generatePseudoRandomSalt;
  static noop = utils.noop;
  static toChecksumAddress = utils.toChecksumAddress;

  /* On Metamask, provider.host is undefined
  Force users to provide host url */
  constructor(
    provider,
    { networkId, addresses = Addresses.getAddresses(networkId) } = {}
  ) {
    if (!networkId) throw new Error(Errors.NoNetworkId);

    assert.isWeb3Provider("provider", provider);
    this.web3 = new Web3(provider);
    this.addresses = addresses;
    this.networkId = networkId;
    switch (networkId) {
      case 1:
        this.networkName = "mainnet";
        this.etherscanURL = "https://etherscan.io/";
        break;
      case 3:
        this.networkName = "ropsten";
        this.etherscanURL = "https://ropsten.etherscan.io/";
        break;
      case 4:
        this.networkName = "rinkeby";
        this.etherscanURL = "https://rinkeby.etherscan.io/";
        break;
      case 42:
        this.networkName = "kovan";
        this.etherscanURL = "https://kovan.etherscan.io/";
        break;
      default:
        this.networkName = "local";
        this.etherscanURL = "";
        break;
    }
  }

  static getLoanOrderHashHex(order) {
    utils.doesConformToSchema("loanOrder", order, schemas.loanOrderSchema);
    const orderHashHex = utils.getLoanOrderHashHex(order);
    return orderHashHex;
  }
  getLoanOrderHashAsync = async props =>
    utils.getLoanOrderHashAsync(this, props);

  static isValidSignature = props => signature.isValidSignature(props);

  isValidSignatureAsync = async props =>
    signature.isValidSignatureAsync(this, props);

  signOrderHashAsync = async (...props) =>
    signature.signOrderHashAsync(this, ...props);

  setAllowance = (...props) => allowance.setAllowance(this, ...props);

  setAllowanceUnlimited = props =>
    this.setAllowance({
      ...props,
      amountInBaseUnits: constants.UNLIMITED_ALLOWANCE_IN_BASE_UNITS
    });

  resetAllowance = props =>
    this.setAllowance({
      ...props,
      amountInBaseUnits: new BigNumber(0)
    });

  getAllowance = async (...props) => allowance.getAllowance(this, ...props);

  getBalance = async ({ tokenAddress, ownerAddress }) => {
    assert.isETHAddressHex("ownerAddress", ownerAddress);
    assert.isETHAddressHex("tokenAddress", tokenAddress);

    const tokenContract = await utils.getContractInstance(
      this.web3,
      EIP20.abi,
      tokenAddress
    );
    const balance = await tokenContract.methods.balanceOf(ownerAddress).call();
    return new BigNumber(balance);
  };

  getTokenList = async () => tokenRegistry.getTokenList(this);

  getOracleList = async () => oracles.getOracleList(this);
  isTradeSupported = async (...props) =>
    oracles.isTradeSupported(this, ...props);

  takeLoanOrderAsLender = (...props) =>
    fill.takeLoanOrderAsLender(this, ...props);

  takeLoanOrderAsTrader = (...props) =>
    fill.takeLoanOrderAsTrader(this, ...props);

  getInitialCollateralRequired = async (...props) =>
    fill.getInitialCollateralRequired(this, ...props);

  getSingleOrder = async (...props) =>
    orderHistory.getSingleOrder(this, ...props);
  getOrders = async (...props) =>
    orderHistory.getOrders(this, ...props);
  getSingleLoan = async (...props) =>
    orderHistory.getSingleLoan(this, ...props);
  getLoansForLender = async (...props) =>
    orderHistory.getLoansForLender(this, ...props);
  getLoansForTrader = async (...props) =>
    orderHistory.getLoansForTrader(this, ...props);

  transferToken = (...props) => transfer.transferToken(this, ...props);

  tradePositionWith0x = (...props) => trade.tradePositionWith0x(this, ...props);
  tradePositionWithOracle = (...props) =>
    trade.tradePositionWithOracle(this, ...props);

  changeCollateral = (...props) => loanHealth.changeCollateral(this, ...props);
  depositCollateral = (...props) =>
    loanHealth.depositCollateral(this, ...props);
  withdrawExcessCollateral = (...props) =>
    loanHealth.withdrawExcessCollateral(this, ...props);
  getProfitOrLoss = (...props) => loanHealth.getProfitOrLoss(this, ...props);
  withdrawProfit = (...props) => loanHealth.withdrawProfit(this, ...props);

  closeLoan = (...props) => loanHealth.closeLoan(this, ...props);
  payInterest = (...props) => loanHealth.payInterest(this, ...props);

  requestFaucetToken = (...props) => utils.requestFaucetToken(this, ...props);

  getActiveLoans = (...props) => bounty.getActiveLoans(this, ...props);
  getMarginLevels = (...props) => bounty.getMarginLevels(this, ...props);
  liquidateLoan = (...props) => bounty.liquidateLoan(this, ...props);

  wrapEth = (...props) => weth.wrapEth(this, ...props);
  unwrapEth = (...props) => weth.unwrapEth(this, ...props);
}
