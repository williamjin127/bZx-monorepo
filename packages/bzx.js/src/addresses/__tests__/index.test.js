import Web3 from "web3";
import { BZxJS } from "../../core/index";
import * as Errors from "../../core/constants/errors";

describe("addresses", () => {
  test("should return kovan testnet addresses for networkId 42", async () => {
    const networkUrl = "https://kovan.infura.io";
    const networkId = 42;
    const provider = new Web3.providers.HttpProvider(networkUrl);
    const bZxJS = new BZxJS(provider, { networkId });

    expect(bZxJS.addresses).toMatchSnapshot();
  });

  test("should return rinkeby testnet addresses for networkId 4", async () => {
    const networkUrl = "http://localhost:8545";
    const networkId = 4;
    const provider = new Web3.providers.HttpProvider(networkUrl);
    const bZxJS = new BZxJS(provider, { networkId });

    expect(bZxJS.addresses).toMatchSnapshot();
  });

  test("should return local testnet addresses for all other networkIds", async () => {
    const networkUrl = "http://localhost:8545";
    const networkId = 50;
    const provider = new Web3.providers.HttpProvider(networkUrl);
    const bZxJS = new BZxJS(provider, { networkId });

    expect(bZxJS.addresses).toMatchSnapshot();
  });

  test("should throw error for no networkId", async () => {
    expect(() => {
      const networkUrl = "http://localhost:8545";
      const provider = new Web3.providers.HttpProvider(networkUrl);
      // eslint-disable-next-line no-unused-vars
      const bZxJS = new BZxJS(provider, { networkId: null });
    }).toThrow(Errors.NoNetworkId);
  });
});
