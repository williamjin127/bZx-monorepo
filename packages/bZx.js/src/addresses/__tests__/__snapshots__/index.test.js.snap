// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addresses should return kovan testnet addresses for networkId 42 1`] = `
Object {
  "BZx": "0x350cbbb706d1acd3bc9f920401e11db33c7e9c74",
  "BZxOracle": "0x5859d1b2b9bd6757e82bce6bc69bb03ffa611e60",
  "BZxTo0x": "0x3da8716483029e0ca72e97e9c5c432920d9bacd1",
  "BZxToken": "0xe8b6a7fa1976ba6c2d3dd81f063eb25d521186bb",
  "BZxVault": "0x93ebdf50a085fda6a1186b2bd52c9e1f045d3248",
  "EIP20": "",
  "OracleInterface": "",
  "OracleRegistry": "0x0000000000000000000000000000000000000000",
  "TestNetFaucet": "0x86bb133db6ba1adc6c664d0d8dc1b34b1a12921a",
  "TokenRegistry": "0x16962232ec2fad399f0d7bee5b8bc60c99991bc6",
  "TokenTransferProxy": "0x087eed4bc1ee3de49befbd66c662b434b15d49d4",
  "WETH": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
  "ZRXToken": "0x6ff6c0ff1d68b964901f986d4c9fa3ac68346570",
}
`;

exports[`addresses should return local testnet addresses for all other networkIds 1`] = `
Object {
  "BZx": "0x8726c7414ac023d23348326b47af3205185fd035",
  "BZxOracle": "0x0d8b0dd11f5d34ed41d556def5f841900d5b1c6b",
  "BZxTo0x": "0x32eecaf51dfea9618e9bc94e9fbfddb1bbdcba15",
  "BZxToken": "0x04b5dadd2c0d6a261bfafbc964e0cac48585def3",
  "BZxVault": "0xb7c9b454221e26880eb9c3101b3295ca7d8279ef",
  "EIP20": "",
  "OracleInterface": "",
  "OracleRegistry": "0x6000eca38b8b5bba64986182fe2a69c57f6b5414",
  "TestNetFaucet": "0x609acc8b356894a937fc58f3411f9528de96ecb1",
  "TestToken0": "0xb48e1b16829c7f5bd62b76cb878a6bb1c4625d7a",
  "TestToken1": "0xc4cc602a7345518d0b7a84049d4bc8575ebf3398",
  "TestToken2": "0xe704967449b57b2382b7fa482718748c13c63190",
  "TestToken3": "0xa4b3e1659c473623287b2cc13b194705cd792525",
  "TestToken4": "0x2c530e4ecc573f11bd72cf5fdf580d134d25f15f",
  "TestToken5": "0x72d5a2213bfe46df9fbda08e22f536ac6ca8907e",
  "TestToken6": "0x2ebb94cc79d7d0f1195300aaf191d118f53292a8",
  "TestToken7": "0x5315e44798395d4a952530d131249fe00f554565",
  "TestToken8": "0xdff540fe764855d3175dcfae9d91ae8aee5c6d6f",
  "TestToken9": "0xc1be2c0bb387aa13d5019a9c518e8bc93cb53360",
  "TokenRegistry": "0x8a063452f7df2614db1bca3a85ef35da40cf0835",
}
`;

exports[`addresses should return rinkeby testnet addresses for networkId 4 1`] = `
Object {
  "BZx": "0x5a5f9a466c4b43fd3eafab2c5d971168fbeb90b8",
  "BZxOracle": "0x014ede18fc693ca54838807aa9e7a26621ca0512",
  "BZxTo0x": "0x797350e9c7dc55d02142105d4a35dde8b9a4e2f6",
  "BZxToken": "0x14823db576c11e4a54ca9e01ca0b28b18d3d1187",
  "BZxVault": "0xe020ebe52e05247a504246526d371ab60ddb601a",
  "EIP20": "",
  "OracleInterace": "",
  "OracleRegistry": "0x2cf2e5f741cea951283524cd68126f2123881065",
  "TestNetFaucet": "0x1ce05b8e436af8be884c60a41127ca01a922ce3d",
  "TokenRegistry": "0x4db8a61f9cd0cf4998aa4612dd612ab4f4f5a730",
  "TokenTransferProxy": "0x4e9aad8184de8833365fea970cd9149372fdf1e6",
  "WETH": "0xc778417e063141139fce010982780140aa0cd5ab",
  "ZRXToken": "0xa8e9fa8f91e5ae138c74648c9c304f1c75003a8d",
}
`;
